### PR TITLE
config: fix osd_crush_initial_weight

### PR DESCRIPTION
**Pull request by JiYou**
1. Fix missing config osd_crush_initial_weight in config_opts.h.
2. Add related docs for osd_crush_initial_weight.

### CODE REVIEW

File: src/common/config_opts.h [hunk 1]
```diff
@@ -550,6 +550,7 @@ OPTION(osd_client_message_cap, OPT_U64, 100)              // num client messages
 OPTION(osd_pg_bits, OPT_INT, 6)  // bits per osd
 OPTION(osd_pgp_bits, OPT_INT, 6)  // bits per osd
 OPTION(osd_crush_chooseleaf_type, OPT_INT, 1) // 1 = host
+OPTION(osd_crush_initial_weight, OPT_DOUBLE, 0) // 0 means default value for osd added into crushmap.
```
Comment (tchaikov) [inline src/common/config_opts.h]: this option is not read by ceph code. so i am sure we need to add it here.

File: doc/rados/configuration/pool-pg-config-ref.rst [hunk 2]
```diff
@@ -66,6 +66,13 @@ Ceph configuration file.
 :Default: ``1``. Typically a host containing one or more Ceph OSD Daemons.
 
 
+``osd crush initial weight``
+:Description: The initial crush weight for the new osd just added into crushmap.
+:Type: Double
+:Default: ``0``. The initial crush weight for the new osd added into crushmap, by
```
Comment (tchaikov) [inline doc/rados/configuration/pool-pg-config-ref.rst]: this is not accurate, the default value is calculated using the size of the osd data volume . it's something like "sizeof($volume)/1TB".

**Comment by JiYou** [comment]
@tchaikov thanks for your helpful suggestion.
I've checked ceph-osd-prestart.sh which acts like your comment.
thanks very much.

**Comment by tchaikov** [comment]
@JiYou and we don't need to change the config_opt.h.

> this option is not read by ceph code. so i am not sure we need to add it here.

**Comment by xiaoxichen** [comment]
@tchaikov , would the config_opt.h act as the collection of config knobs?  I mean, as we don't have a centrolized doc listing the config parameters, config_opt.h somehow act like such doc for some of us. Otherwise it would be too hard for even a experienced user to know this parameters.

**Comment by xiaoxichen** [comment]
but yes, it's not used by ceph C code

**Comment by tchaikov** [comment]
> config_opt.h somehow act like such doc for some of us. 

@xiaoxichen good point, if you insist, i guess we can keep it, but as a comment. what do you think?

**Comment by JiYou** [comment]
@tchaikov @xiaoxichen Thanks for your suggestions. I will check and change it.

**Comment by xiaoxichen** [comment]
@tchaikov something like :

// This parameters is not consumed by ceph but the upstart scripts
// OPTION(osd_crush_initial_weight, OPT_DOUBLE, 0) // the initial weight for newly added OSD.

Works for you ?

**Comment by tchaikov** [comment]
@xiaoxichen yeah!

**Comment by JiYou** [comment]
@tchaikov @xiaoxichen many thanks for you guys' suggestions.

File: src/common/config_opts.h [hunk 3]
```diff
@@ -550,6 +550,8 @@ OPTION(osd_client_message_cap, OPT_U64, 100)              // num client messages
 OPTION(osd_pg_bits, OPT_INT, 6)  // bits per osd
 OPTION(osd_pgp_bits, OPT_INT, 6)  // bits per osd
 OPTION(osd_crush_chooseleaf_type, OPT_INT, 1) // 1 = host
+// This parameter is not consumed by ceph C code but the upstart scripts.
+OPTION(osd_crush_initial_weight, OPT_DOUBLE, 0) // the initial weight is for newly added osds.
```
Comment (xiaoxichen) [inline src/common/config_opts.h]: Also comment this line out.

Comment (JiYou) [inline src/common/config_opts.h]: OK

File: doc/rados/configuration/pool-pg-config-ref.rst [hunk 4]
```diff
@@ -164,3 +172,4 @@ Ceph configuration file.
 
 .. _pool: ../../operations/pools
 .. _Monitoring OSDs and PGs: ../../operations/monitoring-osd-pg#peering
+.. _Weighting Bucket Items: ../../operations/crush-map.rst#Weighting Bucket Items
```
Comment (tchaikov) [inline doc/rados/configuration/pool-pg-config-ref.rst]: i am not sure if this works
> crush-map.**rst**#Weighting Bucket Items

Comment (JiYou) [inline doc/rados/configuration/pool-pg-config-ref.rst]: I will check.

File: doc/rados/configuration/pool-pg-config-ref.rst [hunk 5]
```diff
@@ -164,3 +172,4 @@ Ceph configuration file.
 
 .. _pool: ../../operations/pools
 .. _Monitoring OSDs and PGs: ../../operations/monitoring-osd-pg#peering
+.. _Weighting Bucket Items: ../../operations/crush-map#Weighting Bucket Items
```
Comment (tchaikov) [inline doc/rados/configuration/pool-pg-config-ref.rst]: again this does not work. "Weighting Bucket Items" does not point to any anchor in crush-map.rst. you might want to create a hyper link target in that document first, then reference it from here.

i'd suggest you try `./admin/build-doc` and test before submitting the document related change.

**Comment by JiYou** [comment]
@tchaikov Updated the doc and tried with admin/build-doc and & brower.
Now it works. Thanks for your time and patient.

**Pull request merged**